consts/src/lib.rs

```diff
@@ -154,13 +154,27 @@ impl Default for State<Start> {
     }
 }
 
+pub enum AccessError {
+    OutOfBounds,
+}
+
+impl From<AccessError> for EDHOCError {
+    fn from(error: AccessError) -> Self {
+        match error {
+            AccessError::OutOfBounds => EDHOCError::ParsingError,
```

Comment:
So we are basically mapping all `OutOfBounds` errors to `EDHOCError:ParsingError`. This may result in `ParsingError` for non-parsing, i.e. encoding routines when accidentally trying to access the buffer content, right?

In the case of parsing routines when unsanitized input is received, this will properly return ParsingError.

Reply:
Yes, will add a TODO to think of a way to handle it in a more general case. For now it is fine since encoding routines are expected to always work (they return the value itself, and not Ok/Err).

Reply:
But note that you then won't be able to use the `?` syntax and propagate the error.